### DISCUSSION TITLE
Expand using keyword usage

### DISCUSSION THREAD

**Discussion by muhamad** [dotnet/csharplang#5744] [Language Ideas] [answered]
This discussion is about two related ideas regarding extending the usage of `using` keyword.

### `using` alias with generic arguments and constraints

We already have `using` alias on file scope and `global using` on project scope. The idea here is to allow adding generic arguments and generic constraints to aliases, e.g.

File 1.cs
```csharp
interface SomeInterface
{
}
```

File 2.cs
```csharp
// example 1
using MyEnumerable<T> where T : SomeInterface = System.Collection.Generic.IEnumerable<T>;

// example 2
using SomeDictionary<T> where T:struct = System.Collection.Generic.IDictionary<T, SomeInterface>;
```

In the first example we define an alias which have one generic argument which must be of type `SomeInterface`. The 2nd example define a partial alias where we have only one generic argument which must be a value type get passed to a `IDictionary` as first type parameter and we already define the 2nd parameter as `SomeType`.

### exporting an alias
I'm not sure how this would be implemented, but I think something like:
```csharp
export global using SomeDictionary<T> where T:struct = System.Collection.Generic.IDictionary<T, SomeInterface>;
```
We can have an `ExportAttribute` class instead of `export` keyword.

**Comment by HaloFour** [comment, marked as answer]
See: #90 #5579

**Comment by fa1ntl1ght** [comment]
I expect that #5497 will make `using` aliases obsolete
